src/App.tsx

```diff
@@ -15,14 +15,43 @@ import {
 import { AlphaBanner, GweiBar } from '@components/composite';
 import Box from '@mui/material/Box';
 import { useEffect } from 'react';
+import {
+  deleteReferrer,
+  isRefererStored,
+  isValidReferrerStored,
+  isValidReferrerValue,
+  setReferrer,
+} from './utilities/referrer-store/referrer-store';
 
 const App = () => {
+  const [searchParams] = useSearchParams();
+  const searchParamsReferrer = searchParams.get('invitedBy');
+
   useEffect(() => {
     if (process.env.REACT_APP_GTM_CODE) {
       TagManager.initialize({ gtmId: process.env.REACT_APP_GTM_CODE });
     }
   }, []);
 
+  // referrer logic - run everytime params change for invitedBy
+  useEffect(() => {
+    // Earlier, pre-release keys may have used more than 8 characters, but we overwrite these
+    if (!isValidReferrerStored()) {
```

Comment:
Do we need this block of code or would the one on line 50 (with the suggested change to use isValidReferrerStored) be ok on its own?

Reply:
well this is something that happens first. Even if they don't have a search param (missing ?invitedBy=XXX), but have old invalid value in localstorage we will reset it. I tried to split the logic you did and make it more modular.

Reply:
Understood, and fine to keep if you want, but I'm not sure we need to do that. Sufficient to just replace it with something new (or leave it alone if there's nothing new).

Reply:
I will keep it as it is. To my mind cleaning up ensures we have a clean state for the "next feature" we want to build on top of this.